### PR TITLE
fix(47024):fix type of DateTimeFormat and NumberFormat

### LINKED ISSUE #47024: `formatRange[ToParts]` should go `Intl.DateTimeFormat` but instead is in `NumberFormat`
# Bug Report

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions ______ and _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because _______

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=8#code/JIOwLgNgdAcgrgWwEYFMBOAxA9mhBDMKABzSzDIE8iUoAzHfMAJTxAHMUACAem8+NLkwVLgEsAzp1YUA-AEIAsAChlIFAHdOoSLESpMDAgAoAlHUPNWHHn3EALLHAgATTiDKdUUgG55REPCQIMRBOFHEAJgAGCIBGABpOcTB-CE5RUOS8awBmVQ0tcGgAEQIUABVRBBRsXGMzejrLdi5eJIcnVy88X39A4PTQ8Oi4oA)

### 💻 Code

```ts
Intl.NumberFormat.prototype.formatRange // .prototype is any?!

new Intl.NumberFormat().formatRange // should not be available in es2021, still in stage 3
new Intl.DateTimeFormat().formatRange // should be available in es2021
```

### 🙁 Actual behavior

(As the title says)

### 🙂 Expected behavior

(As the comments say)

#45647 was confused about this.

### PR DESCRIPTION
**Pull request by islandryu**
Fixes #47024

### CODE REVIEW

File: src/lib/es2021.intl.d.ts [hunk 1]
```diff
@@ -16,9 +21,4 @@ declare namespace Intl {
         dayPeriod?: "narrow" | "short" | "long";
         fractionalSecondDigits?: 0 | 1 | 2 | 3;
     }
-
-    interface NumberFormat {
```
Comment (saschanaz) [inline src/lib/es2021.intl.d.ts:20]: Could you perhaps move this to esnext.intl.d.ts?

Comment (islandryu) [reply src/lib/es2021.intl.d.ts:20]: Thanks for the review!
I've fixed it.

File: src/lib/esnext.intl.d.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
 declare namespace Intl {
-   // Empty for now
+
+    interface NumberFormat {
+        formatRange(startDate: number | bigint, endDate: number | bigint): string;
+        formatRangeToParts(startDate: number | bigint, endDate: number | bigint): NumberFormatPart[];
```
Comment (sandersn) [inline src/lib/esnext.intl.d.ts:5]: From reading the spec, I see that [the parameter names are `start` and `end`](https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/proposed.html#sec-intl.numberformat.prototype.formatrange).

```suggestion
        formatRange(start: number | bigint, end: number | bigint): string;
        formatRangeToParts(start: number | bigint, end: number | bigint): NumberFormatPart[];
```

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good except the parameter names. I took a long time to review this -- @islandryu I can add my suggestion myself if you want. I'm going to hold this until 4.7 anyway since we're nearly through the 4.6 beta period.

**Comment by islandryu** [comment]
@sandersn 
Thanks for the review!
I've committed the suggestion and renamed the conflict.

**Review by sandersn — APPROVED** [review]

**Pull request merged**